pkg/providers/imagefamily/resolver.go

```diff
@@ -95,25 +100,25 @@ func NewDefaultResolver(_ client.Client, imageProvider *provider, instanceTypePr
 	}
 }
 
-// Resolve fills in dynamic launch template parameters
+// Resolve fills in dynamic launch template parameters.
+// The name "imageFamilyResolver.Resolve()" is potentially misleading here.
+// Suggestion: refactor would help, but this won't be used by PROVISION_MODE=aksmachineapi anyway. May not be worth it.
+// ATTENTION!!!: changes here may NOT be effective on AKS machine nodes (ProvisionModeAKSMachineAPI); See aksmachineinstance.go/aksmachineinstancehelpers.go.
```

Comment:
Agree this is sorta confusing. Also not sure if refactor is worth it, but splitting these specific changes into a smaller PR and/or looking more closely at just these bits may give us a better idea.